src/main/scala/nl.knaw.dans.easy.sword2/DepositHandler.scala

```diff
@@ -66,27 +66,21 @@ object DepositHandler {
       }
     settings
       .tempDir.listFiles().toSeq
-      .filter(_.isDirectory)
-      .filter {
-        d =>
-          getDepositState(d)
-            .map(_ == State.UPLOADED)
-            .recoverWith {
-              case _: Throwable =>
-                log.warn(s"[${ d.getName }] Could not get deposit state. Not putting this deposit on the queue.")
-                Success(false)
-            }.get
-      }.foreach {
-      d =>
-        getContentType(d).map {
-          mimeType =>
-            log.info(s"[${ d.getName }] Scheduling UPLOADED deposit for finalizing.")
-            depositProcessingStream.onNext((d.getName, mimeType))
-        }.recover {
-          case _: Throwable =>
-            log.warn(s"[${ d.getName }] Could not get deposit Content-Type. Not putting this deposit on the queue.")
-        }
-    }
+      .filter(deposit => deposit.isDirectory && filterNonUploadedDeposits(deposit))
```

Comment:
Great job refactoring this mess into something beautiful!!!

Would you mind splitting it up into two `withFilters`. That way we only have one iteration through the sequence.

```suggestion
      .withFilter(_.isDirectory)
      .withFilter(filterNonUploadedDeposits)
```